### PR TITLE
Migrated code from Class based component to Function based component of WikiSelect component

### PR DESCRIPTION
**Pull request by shiven16**
NOTE: Please review the pull request process before opening your first PR: https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/master/CONTRIBUTING.md#pull-request-process

## What this PR does
This pull request refactors the WikiSelect component from a class-based component to a functional component .

##Screenshots

Before:
Pr2 before

After:
Pr2 after

## Open questions and concerns
Please review and provide feedback on my changes. I am committed to addressing any issues promptly.

### CODE REVIEW

**Comment by ragesoss** [comment]
Please include before/after screenshots.

**Comment by shiven16** [comment]
Hey @ragesoss I have uploaded the screenshots for the UI component, could you please review the PR again.

File: app/assets/javascripts/components/common/wiki_select.jsx [hunk 1]
```diff
@@ -11,167 +10,149 @@ import { formatOption } from '../../utils/wiki_utils';
...
+        wiki = wiki.map((w) => { return { value: JSON.parse(w.value), label: w.label }; });
       } else {
-        this.props.onChange(this.props.multi ? [] : {});
+        const value = JSON.parse(wiki.value);
+        wiki = { label: wiki.label, value };
       }
-    };
-
-    // If the component was passed a set of wikis as options, limit the options to those wikis.
-    // Otherwise, allow any wiki to be selected.
```
Comment (ragesoss) [inline app/assets/javascripts/components/common/wiki_select.jsx:121]: What is the reason for not retaining these comments?

Comment (shiven16) [reply app/assets/javascripts/components/common/wiki_select.jsx:121]: Hey @ragesoss sorry for overlooking those comments I have updated my code.

**Pull request merged**